connector/new.go

```diff
@@ -216,9 +216,11 @@ func newPipelinerConnector(parameters Parameters) (connectors.Connector, error)
 }
 
 func newSmartleadConnector(parameters Parameters) (connectors.Connector, error) {
-	return smartlead.NewConnector(
-		smartlead.WithAuthenticatedClient(parameters.AuthenticatedClient),
-	)
+	// Client could directly be set via common methods -
+	// hc, _ = common.NewHeaderAuthHTTPClient(context.TODO())
+	// parameters.AuthenticatedClient = hc
+
+	return smartlead.NewConnector(parameters)
```

Comment:
I reviewed the other PR but will leave a review here.

The options design pattern that was previously used:
( :cry: ) created duplicated files with methods for each connector
( :cry: ) complicated deep connector architecture redesign, where everything had to resolve around this
( :smile: ) provided a strict list of parameters, the code tells what the connector needs.

Unified parameters across connectors (this PR):
( :cry: ) doesn't tell by autocomplete which properties are important and which are irrelevant (ex: connector may not have metadata), when initializing connector it is not clear what properties I must provide (validation errors would guide in this case)
( :neutral_face: ) creates a central place for parameter definition. From the surface connectors look similar so it makes sense. I think there is no expectation for it to rapidly change so it is neither positive nor negative.
( :smile: ) generalizaing deep connector architecture will be easier, there is no need to use go generics by every place that depends on parameters.